Lombiq.Hosting.MediaTheme.Deployer/Program.cs

```diff
@@ -171,13 +171,21 @@ private static void RunOptions(CommandLineOptions values)
         {
             var fileName = file.Name;
 
-            if (areLiquidFiles)
+            // Files should follow the template name convention (with underscores only), not the template file name
+            // convention (with dots and hyphens). Note that these don't just differ in the interchangeable characters
+            // but may also differ in the order of sections. E.g.
+            // [Stereotype]_[DisplayType]__[PartType]__[PartName]__[DisplayMode]_Display is
+            // "Widget_Summary__ServicePart__Services__CustomMode_Display" as template name but
+            // "Widget-ServicePart-Services-CustomMode.Display.Summary.liquid" as a file name (note e.g. "Summary" being
+            // in a different location).
+            var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(fileName);
+            if (areLiquidFiles && (fileNameWithoutExtension.Contains('.') || fileNameWithoutExtension.Contains('-')))
             {
-                var fileNameWithoutExtension = fileName[..fileName.LastIndexOf(
-                    LiquidFileExtension,
-                    StringComparison.InvariantCulture)];
-
-                fileName = fileNameWithoutExtension.Replace('.', '_').Replace("_", "__") + LiquidFileExtension;
+                throw new InvalidOperationException(
```

Comment:
If you have it otherwise, then yes, but this was the case already, see [the docs before these changes](https://github.com/Lombiq/Hosting-Media-Theme#:~:text=Shape%20template%20file%20names%20must%20be%20valid%20shape%20types), just not enforced. The original code here worked with the assumption that the only difference between the template names and file names are dots vs underscores and single underscores vs double underscores. However, this is not the case, the order of identifiers in the name is also different (what I explained in the updated docs).